store/store.js

```diff
@@ -0,0 +1,63 @@
+import { createStore, action, thunk, computed } from 'easy-peasy';
+import config from '../config';
+import apiUtils from '../api/api';
+
+const storeState = {
+  currentUser: null,
+  loginError: '',
+  signUpError: '',
+};
+
+const getters = {
+  /*
+  enabledAndHistoricalBankAccounts: computed(state => (
+    state.bankAccounts.filter((bankAccount => bankAccount.active && bankAccount.withMatches))
+  )),
+  */
+};
+
+const storeActions = {
+  addCurrentUser: action((state, payload) => {
+    state.currentUser = payload;
+  }),
+  setLoginError: action((state, payload) => {
+    state.loginError = payload;
+  }),
+  setSignUpError: action((state, payload) => {
+    state.signUpError = payload;
+  }),
+};
+
+const storeThunks = {
+  sendCredentials: thunk(async (actions, payload) => {
+    const url = payload.logIn ? config.endpoints.users.logIn : config.endpoints.users.createUser;
+    await apiUtils.api({
+      method: 'post',
+      url,
+      data: payload,
+    }).then((resp) => {
+      if (resp.status === apiUtils.statusCodes.ok || resp.status === apiUtils.statusCodes.created) {
+        const user = resp.data.data.attributes;
+        actions.addCurrentUser(user);
+        apiUtils.api.defaults.headers = { 'Accept': 'application/json',
+          'Content-Type': 'application/json', 'X-User-Mail': user.email,
+          'X-User-Token': user.authentication_token };
+      }
+    }).catch((error) => {
+      if (payload.logIn) {
+        actions.setLoginError(error.response.data.message);
+      } else {
+        actions.setSignUpError(error.response.data.message);
+      }
+    });
+  }),
```

Comment:
sería bueno mover está lógica a un archivo `api/sessions.js` que hace un` default export` que después importamos como `seesionsApi`

me imagino que los `thunks` quedarían algo así:
```suggestion
login: thunk(async (actions, payload) => {
  sessionsApi.login(payload)
    .then(res => {
        actions.setUserAndApiHeaders(res)
    }).catch((error) => {
        actions.setLoginError(error.response.data.message);
    });
  }),
  signUp: thunk(async (actions, payload) => {
  sessionsApi.signUp(payload)
    .then(res => {
        actions.setUserAndApiHeaders(res)
    }).catch((error) => {
        actions.setSignUpError(error.response.data.message);
    });
  }),
```